lib/tessel/setup.js

```diff
@@ -0,0 +1,139 @@
+
+var osenv = require('osenv');
+var fs = require('fs-extra');
+var keygen = require('ssh-keygen');
+
+var Tessel = require('./tessel');
+var commands = require('./commands')
+
+var dir = osenv.home() + '/.tessel';
+var filename = 'id_rsa';
+var filepath = dir + '/' + filename;
+
+var tessel;
+
+Tessel.prototype.setupTessel = function (opts) {
+  var self = this;
+  // Check if local .tessel file has keypair, if not, put it there
+  setupLocal(function () {
+    // Authorize Tessel for SSH
+    authTessel(self);
+  });
+}
+
+// Make sure local computer is set up to authorize with Tessel
+function setupLocal (callback) {
+  // Check if keypair already in place
+  if(!(fs.existsSync(filepath) && fs.existsSync(filepath + '.pub'))) {
+    console.log('Creating public and private keys for Tessel authentication...');
+
+    // Generate SSH key
+    keygen({}, function(err, out){
+      if(err) return console.log('Something went wrong: '+err);
+      var privateKey = out.key;
+      var publicKey = out.pubKey;
+
+      // Make sure dir exists
+      fs.ensureDir(dir, function (err) {
+        err && console.log(err);
+
+        filesWritten = 0;
+
+        // Put SSH keys for Tessel in that folder
+        fs.writeFile(filepath + '.pub', publicKey, function (err) {
+          err && console.log(err);
+          fs.chmodSync(filepath + '.pub', '600');
+          console.log('Public key written.');
+          filesWritten++;
+          if (filesWritten == 2) {
+            callback && callback();
+          }
+        });
+        fs.writeFile(filepath, privateKey, function (err) {
+          err && console.log(err);
+          fs.chmodSync(filepath, '600');
+          console.log('Private key written.');
+          filesWritten++;
+          if (filesWritten == 2) {
+            callback && callback();
+          }
+        });
+      });
+    });
+  } else {
+    callback && callback();
+  }
+}
+
+// Put the specified SSH key in Tessel's auth file
+function authTessel (tessel) {
+  console.log('Authenticating Tessel with public key...');
+
+  var sourceFile = filepath + '.pub';
+  var authFile = '/etc/dropbear/authorized_keys';
+
+  // Make sure Tessel has the authFile
+  checkAuthFile(tessel, authFile, function () {
+    // Read the public key
+    fs.readFile(filepath + '.pub', {encoding: 'utf-8'}, function (err, pubKey) {
+      // See if the public key is already in the authFile
+      checkIfKey(tessel, authFile, pubKey, function () {
+        // Copy pubKey into authFile
+        copyKey(tessel, authFile, pubKey);
+      });
+    });
+  });
+}
+
+function checkAuthFile (tessel, authFile, callback) {
+  tessel.connection.exec(commands.ensureFile(authFile), function (err, streams) {
+    streams.stderr.pipe(process.stderr);
+    streams.stdout.on('end', function () {
+      callback && callback();
+    });
+  });
+}
+
+function checkIfKey (tessel, authFile, pubKey, callback) {
```

Comment:
Again, what this function does is unclear. I recommend making it more verbose `checkIfKeyExists`.